simtools/simulator.py

```diff
@@ -0,0 +1,740 @@
+        """
+        Fill in the results dict without calling submit.
+
+        Parameters
+        ----------
+        inputFileList: str or list of str
+            Single file or list of files of shower simulations.
+
+        """
+        inputFileList = self._enforceListType(inputFileList)
+
+        for file in inputFileList:
+            run = self._guessRunFromFile(file)
+            self._fillResults(file, run)
+            self.runs.append(run)
+
+    def submit(self,
+               inputFileList=None,
+               submitCommand=None,
+               extraCommands=None,
+               test=False):
+        """
+        Submit a run script as a job. The submit command can be given by \
+        submitCommand or it will be taken from the config.yml file.
+
+        Parameters
+        ----------
+        inputFileList: str or list of str
+            Single file or list of files of shower simulations.
+        submitCommand: str
+            Command to be used before the script name.
+        extraCommands: str or list of str
+            Extra commands to be added to the run script before the run command,
+        test: bool
+            If True, job is not submitted.
+
+        """
+
+        subCmd = submitCommand if submitCommand is not None else cfg.get("submissionCommand")
+        self._logger.info("Submission command: {}".format(subCmd))
+
+        runs_and_files_to_submit = self._get_runs_and_files_to_submit(
+            inputFileList=inputFileList)
+        self._logger.info("Starting submission for {} runs".format(
+            len(runs_and_files_to_submit)))
+
+        for run, file in runs_and_files_to_submit.items():
+
+            runScript = self._simulationRunner.getRunScript(
+                runNumber=run, inputFile=file, extraCommands=extraCommands
+            )
+
+            job_manager = JobManager(submitCommand=subCmd, test=test)
+            job_manager.submit(
+                run_script=runScript,
+                run_out_file=self._simulationRunner.getSubLogFile(runNumber=run, mode=""),
+            )
+
+            self._fillResults(file, run)
+
+    def _get_runs_and_files_to_submit(self, inputFileList=None):
+        """
+        Return a dictionary with run numbers and (if applicable) simulation
+        files. The latter are expected to be given for the simtel simulator.
+
+        Parameters
+        ----------
+        inputFileList: str or list of str
+            Single file or list of files of shower simulations.
+
+        Returns
+        -------
+        runs_and_files: dict
+            dictionary with runnumber as key and (if availble) simulation
+            file name as value
+
+        """
+
+        _runs_and_files = {}
+
+        if self.simulator == 'simtel':
+            _file_list = self._enforceListType(inputFileList)
+            for file in _file_list:
+                _runs_and_files[self._guessRunFromFile(file)] = file
+        else:
```

Comment:
Should check if it's CORSIKA here in case in the future we have something else (like CARE or SST-CAM simulator).

Reply:
OK!